### PR TITLE
Upgrade Chart.js engine from v1.0.1 to v2.3.0

### PR DESCRIPTION
**Pull request by DL6ER**
Changes proposed in this pull request:

- Upgrade Chart.js from version 1.0.1 (close to ancient) to 2.3.0 (current version).

As the syntax changed quite significantly, a major edit has been necessary. However, there is no conflict against the current development branch and an automatic merge is easily possible.

I think this upgrade is long overdue. I need the current version for my planed enhancements to the web interface. With the upgrade we can now use advanced chart options (like adding a second y axis, needed to make progress on issue #138).

Before this pull request:

old

And afterwards:

new

@pi-hole/dashboard

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -65,7 +65,7 @@
         </div>
         <div class="box-body">
           <div class="chart">
-            <canvas id="queryOverTimeChart" style="height: 247px; width: 466px;" width="932" height="494"></canvas>
```
Comment (DL6ER) [inline index.php]: This seems to me unnecessarily complicated. I substituted it with a much simpler expression.

File: js/pihole/index.js [hunk 2]
```diff
@@ -50,32 +32,74 @@ $(document).ready(function() {
     };
     var animate = false;
     var ctx = document.getElementById("queryOverTimeChart").getContext("2d");
-    timeLineChart = new Chart(ctx).Line(chartData,
-        {
-            pointDot : false,
-            legendTemplate : "<ul class=\"<%=name.toLowerCase()%>-legend\"><% for (var i=0; i<datasets.length; i++){%><li><span style=\"background-color:<%=datasets[i].strokeColor%>\"></span><%if(datasets[i].label){%><%=datasets[i].label%><%}%></li><%}%></ul>",
-            animation : animate
+    timeLineChart = new Chart(ctx, {
```
Comment (DL6ER) [inline js/pihole/index.js]: The properties of the "timeLineChart" have now to be set in a very different format.

File: js/pihole/index.js [hunk 3]
```diff
@@ -110,11 +134,39 @@ function updateSummaryData(runOnce) {
 
 function updateQueriesOverTime() {
     $.getJSON("api.php?overTimeData", function(data) {
+        // Add data for each hour that is available
         for (hour in data.ads_over_time) {
-            timeLineChart.addData([data.domains_over_time[hour], data.ads_over_time[hour]], hour + ":00");
+            // Add x-axis label
```
Comment (DL6ER) [inline js/pihole/index.js]: The *.addData routine is not available anymore. I now do it like this to achieve the same behavior.

**Comment by PromoFaux** [comment]
@DL6ER thanks for this, but please submit pull requests from and against the `devel` branch, and not `master`, thanks!

**Comment by DL6ER** [comment]
Whoops, I haven't seen that it went against master. This shouldn't have happened. I'll open another pull request.

**Pull request closed without merging**